### ISSUE TITLE
Modal: should have the `open` attribute by default and an label for the "close" button

### ISSUE THREAD

**Issue by tassoevan** [RocketChat/fuselage#1387]
**Description**
Right now `Modal` doesn't render a `dialog` element with the `open` attribute. This is a a11y problem, since we only render the `<dialog>` markup when the modal is, in fact, open, but accessibility tools can't read it as such.
Additionally, the "close" button doesn't look accessible by default, since is missing an default `aria-label` value. At least `Modal.Close` must have an mandatory prop that maps to `aria-label` or `aria-labelledby`.

**Steps to reproduce**
1. Write a unit test using Testing Library to render a `<Modal />`
2. Add the expectation `expect(screen.getByRole('dialog')).toBeInTheDocument()`;
3. Run the test.

**Insert an image/gif**
